pkg/manager/impl/execution_manager.go

```diff
@@ -255,8 +276,31 @@ func assignResourcesIfUnset(ctx context.Context, identifier *core.Identifier,
 		}
 		logger.Debugf(ctx, "Setting 'memory' for [%+v] to %s", identifier, platformValues.Memory)
 		resourceEntries = append(resourceEntries, memoryResource)
+		resourceIndices.memoryIndex = len(resourceEntries) - 1
+	}
+	return resourceEntries, resourceIndices
+}
+
+// This func makes sure that between all the default values substitution that the requests for a specific resource
+// never exceed the limits.
+func finalizeTaskResources(ctx context.Context, task *core.CompiledTask,
+	resourceRequestIndex, resourceLimitIndex int, resourceName string) error {
+	requested, err := resource.ParseQuantity(task.Template.GetContainer().Resources.Requests[resourceRequestIndex].Value)
+	if err != nil {
+		return errors.NewFlyteAdminErrorf(codes.InvalidArgument, "Unable to parse requested cpu quantity [%s]: [%v]",
+			task.Template.GetContainer().Resources.Requests[resourceRequestIndex].Value, err)
+	}
+	limit, err := resource.ParseQuantity(task.Template.GetContainer().Resources.Limits[resourceLimitIndex].Value)
+	if err != nil {
+		return errors.NewFlyteAdminErrorf(codes.InvalidArgument, "Unable to parse cpu limit quantity [%s]: [%v]",
+			task.Template.GetContainer().Resources.Limits[resourceLimitIndex].Value, err)
+	}
+	if limit.Cmp(requested) < 0 {
```

Comment:
This behavior needs to be documented somewhere... Somebody someday will wonder why it didn't happen the other way around and we need to be able to point to a doc that clearly states that.

Reply:
Good point. And as said in my comment above that the corresponding unit test needs to reflect this behavior explicitly.